core/templates/domain/feature_gating/ClientContextObjectFactory.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating ClientContext domain objects.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+export interface ClientContextBackendDict {
+  'client_type': string;
+  'browser_type': string;
+  'app_version'?: string;
+  'user_locale': string;
+}
+
+export class ClientContext {
```

Comment:
Please add documentation for all new exported classes to explain what they are, and how they should be used.

Reply:
Done, added docs for `ClientContext`, `FeatureFlagResult`, `PlatformParameter`, `PlatformParameterRule`, `PlatformParameterFilter`.